bn-apps/memberships-management/membership-service/src/main/kotlin/net/corda/businessnetworks/membership/bno/ActivateMembershipFlow.kt

```diff
@@ -20,14 +21,12 @@ import net.corda.core.transactions.TransactionBuilder
  * @param membership membership state to be activated
  */
 @InitiatingFlow
-class ActivateMembershipFlow(val membership : StateAndRef<Membership.State>) : FlowLogic<SignedTransaction>() {
+@StartableByRPC
+class ActivateMembershipFlow(val membership : StateAndRef<Membership.State>) : BusinessNetworkAwareFlow<SignedTransaction>() {
 
     @Suspendable
     override fun call() : SignedTransaction {
-        // Only BNO should be able to start this flow
-        if (ourIdentity != membership.state.data.bno) {
-            throw FlowException("Our identity has to be BNO")
-        }
+        checkWeAreTheBNOOnThisMembership(membership.state.data)
```

Comment:
Looks too long :) Maybe lets rename to something shorter like verifyThatWeAreBNO()?

Reply:
ok